sacred/utils.py

```diff
@@ -36,6 +39,8 @@
 
 PYTHON_IDENTIFIER = re.compile("^[a-zA-Z_][_a-zA-Z0-9]*$")
 
+PathType = Union[str, bytes, Path]
```

Comment:
Could we use `os.PathLike` instead?

Reply:
I can't actually find a clear answer to this question. I though about looking into the python stub file for os.path but it's not actually informative. https://github.com/python/typeshed/blob/edee7cfb3cc5af16afa557f1279c22ee6663a61f/stdlib/3/os/path.pyi#L14 . I'm open to any idea.

From simple experiments, it seems that 
`isinstance(pathlib.Path('.'), os.PathLike)` returns true but 
`isinstance('dodo', os.PathLike)` returns False.

Any guidance is welcome.

Reply:
True. Also this discussion https://github.com/python/typing/issues/402 seems to state that there is the idea to have a protocol class for this at some time, but currently it is something different. I would then probably use 
`PathType = Union[str, bytes, os.PathLike]`

Reply:
Should we slowly migrate the internals of sacred to pathlib? That would make the code easier to read. We can do so while staying backward compatible of course.

Reply:
os.pathlike is not available for python 3.5. That solves the issue. Going back to pathlib.

Reply:
Alright, migrating sacred internals to pathlib sounds like a very good idea!